### PR TITLE
chore: add github action to generate API index

### PR DESCRIPTION
**Pull request by dwsupplee**
This action will regenerate the API index introduced by https://github.com/googleapis/googleapis/pull/646. It is triggered on each commit coming in to master. 

As [the generator](https://github.com/googleapis/googleapis-api-index-generator/) develops we will tag a stable reason and pin to that in this action.

### CODE REVIEW

File: .github/workflows/generate_api_index.yaml
Comment (danoscarmike) [inline .github/workflows/generate_api_index.yaml:17]: Should this be `googleapis/googleapis-api-index-generator`?

**Review by miraleung — CHANGES_REQUESTED** [review]

File: .github/workflows/generate_api_index.yaml
Comment (dwsupplee) [reply .github/workflows/generate_api_index.yaml:17]: Yep! I opened https://github.com/googleapis/googleapis-api-index-generator/pull/5 to address a few minor issues. We can either merge this now to get this action working right away and update this target after, or wait until Jon has a moment to review and merge.

Comment (danoscarmike) [reply .github/workflows/generate_api_index.yaml:17]: Let's give Jon overnight

File: .github/workflows/generate_api_index.yaml
Comment (danoscarmike) [reply .github/workflows/generate_api_index.yaml:26]: LGTM

File: .github/workflows/generate_api_index.yaml
Comment (dwsupplee) [reply .github/workflows/generate_api_index.yaml]: Updated!

**Review by danoscarmike — APPROVED** [review]

**Pull request merged**
